warehouse/query-core/src/main/java/datawave/query/discovery/DiscoveryIterator.java

```diff
@@ -17,141 +18,318 @@
 import com.google.common.collect.ArrayListMultimap;
 import com.google.common.collect.Multimap;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import datawave.ingest.protobuf.Uid;
+import datawave.marking.MarkingFunctions;
+import datawave.query.Constants;
 
 public class DiscoveryIterator implements SortedKeyValueIterator<Key,Value> {
+
     private static final Logger log = Logger.getLogger(DiscoveryIterator.class);
+    private static final MarkingFunctions markingFunctions = MarkingFunctions.Factory.createMarkingFunctions();
 
-    private Key tk;
-    private Value tv;
-    private SortedKeyValueIterator<Key,Value> itr;
+    private Key key;
+    private Value value;
+    private SortedKeyValueIterator<Key,Value> iterator;
     private boolean separateCountsByColVis = false;
     private boolean showReferenceCount = false;
     private boolean reverseIndex = false;
+    private boolean sumCounts = false;
 
     @Override
     public DiscoveryIterator deepCopy(IteratorEnvironment env) {
-        DiscoveryIterator i = new DiscoveryIterator();
-        i.itr = itr.deepCopy(env);
-        return i;
+        DiscoveryIterator copy = new DiscoveryIterator();
+        copy.iterator = iterator.deepCopy(env);
+        return copy;
     }
 
     @Override
     public void next() throws IOException {
-        tk = null;
-        tv = null;
+        this.key = null;
+        this.value = null;
 
-        while (itr.hasTop() && tk == null) {
-            Multimap<String,TermInfo> terms = aggregateDate();
+        while (iterator.hasTop() && key == null) {
+            // Get the entries to aggregate.
+            Multimap<String,TermEntry> terms = getTermsByDatatype();
             if (terms.isEmpty()) {
-                if (log.isTraceEnabled())
-                    log.trace("Couldn't aggregate index info; moving onto next date/field/term if data is available.");
-                continue;
+                log.trace("Couldn't aggregate index info; moving onto next date/field/term if data is available.");
             } else {
-                if (log.isTraceEnabled())
-                    log.trace("Received term info multimap of size [" + terms.size() + "]");
-                ArrayList<DiscoveredThing> things = newArrayList(
-                                filter(transform(terms.asMap().values(), new TermInfoAggregation(separateCountsByColVis, showReferenceCount, reverseIndex)),
-                                                Predicates.notNull()));
-                if (log.isTraceEnabled())
-                    log.trace("After conversion to discovery objects, there are [" + things.size() + "] term info objects.");
-                if (things.isEmpty()) {
-                    continue;
-                } else {
-                    Pair<Key,Value> top = makeTop(things);
-                    tk = top.getFirst();
-                    tv = top.getSecond();
+                if (log.isTraceEnabled()) {
```

Comment:
Not a dealbreaker, but would it make sense to prune out the trace checks for here and line 73 since it was done in a few other spots?